wazo_call_logd/generator.py

```diff
@@ -91,57 +89,42 @@ def _remove_duplicate_participants(self, call_log):
                 continue
-            raw_attributes.update(**confd_participant)
-            answered = {}
-            if 'answered' in raw_attributes:
-                answered['answered'] = raw_attributes['answered']
-            participant_model = CallLogParticipant(
-                user_uuid=confd_participant['uuid'],
-                line_id=confd_participant['line_id'],
-                tags=confd_participant['tags'],
-                role=raw_attributes['role'],
-                **answered,
-            )
-            call_log.participants.append(participant_model)
-
-    def _update_call_participants_with_their_tags(self, confd, call_log):
-        source_user_uuid = call_log.source_user_uuid
-        destination_user_uuid = call_log.destination_user_uuid
-
-        if source_user_uuid:
-            source_user = confd.users.get(source_user_uuid)
-            source_user_tags = (
-                [tag.strip() for tag in source_user['userfield'].split(',')]
-                if source_user['userfield']
-                else []
-            )
-
-        if destination_user_uuid:
-            destination_user = confd.users.get(destination_user_uuid)
-            destination_user_tags = (
-                [tag.strip() for tag in destination_user['userfield'].split(',')]
-                if destination_user['userfield']
-                else []
-            )
+            raw_attributes.update(**confd_participant._asdict())
+            if confd_participant.uuid in participants_by_uuid:
+                participant = participants_by_uuid.pop(confd_participant.uuid)
+            else:
+                participant = CallLogParticipant(user_uuid=confd_participant.uuid)
+                call_log.participants.append(participant)
 
-        call_log_new_participants = []
-        for participant in call_log.participants:
-            if source_user_uuid:
-                if participant.user_uuid == source_user_uuid:
-                    participant.tags = source_user_tags
-                    participant.line_id = source_user['lines'][0]['id']
-            if destination_user_uuid:
-                if participant.user_uuid == destination_user_uuid:
-                    participant.tags = destination_user_tags
-                    participant.line_id = destination_user['lines'][0]['id']
-            call_log_new_participants.append(participant)
-        call_log.participants = call_log_new_participants
+            participant.line_id = confd_participant.line_id
+            participant.tags = confd_participant.tags
+            participant.role = raw_attributes['role']
+            if 'answered' in raw_attributes:
+                participant.answered = raw_attributes["answered"]
+
+        if participants_by_uuid:
+            # remaining participants identified by CEL interpretation but have no matching channel
+            for uuid, participant in participants_by_uuid.items():
```

Comment:
TL;DR: fetching participants by channel should have priority because of authoritative information on line.

In my mind, there are these cases for participants:
* they have been identified(with their user_uuid) from CEL interpretation
  * they have been reached, so they have an associated channel
  * they have not been reached, so have no associated channel
* they have not been identified from CEL interpretation. they will only be identified from their corresponding channel

However, only if the participant has been reached and has an associated channel can the proper line be associated to the participant. The `find_participant` function first fetches the line from the channel name, then the associated user.
If there is no channel, the `find_participant_by_uuid` fetches the user directly by its `user_uuid`(extracted from CEL interpretation). The implementation provides the first registered line of the user as the value for `line_id`, because that is what was previously done by `_update_call_participants_with_their_tags` for source and destination users. 
This may not be desirable, and `line_id` could be left to `None` for this case.